registry/handlers/app.go

```diff
@@ -102,6 +105,22 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		isCache: config.Proxy.RemoteURL != "",
 	}
 
+	if app.Config.Reporting.NewRelic.LicenseKey != "" {
```

Comment:
I am not entirely sure how I feel about this leaking into the `app` handler. I'm worried adding `newrelic` integration like this may open the floodgates for other commercial integrations and turn this code int something unpleasant...I wonder if we could somehow isolate these parts into dedicated source files 🤔

Reply:
I am going to refactor this to look more like a pluggable middleware. Setup will mostly function in a similar fashion to the storage middlewares, but I am hoping to keep the config under the `reporting` top level key so that any additional instrumentation for New Relic will be able to share it.

Reply:
Sorry, swamped at $DAY_JOB - I like this approach! 😅 I haven't got around properly reviewing yet but should be able to hopefully this week. Based on the first look the only question mark I have is the vanity import of `newrelic` in the main. I feel like we could make it so that we import only `http` middlewares
```
"github.com/distribution/distribution/v3/registry/http/middleware"
```
Or maybe even
```
"github.com/distribution/distribution/v3/registry/http"
```
And those packages would be importing individual middlewares. That way we keep the main clean or 3rd party integrations. WDYT?

Reply:
Thanks for taking a look. I moved the import to `github.com/distribution/distribution/v3/registry/http`.